java/apps/authentication-portal/src/main/webapp/add-security-questions.jsp

```diff
@@ -97,16 +97,28 @@
 </head>
 <body class="login-portal layout authentication-portal-layout">
     <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
-        <layout:component componentName="ProductHeader" >
+        <layout:component componentName="ProductHeader">
             <%-- product-title --%>
             <%
-                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+            File productTitleFile = null;
+            String productTitleFilePath;
+            
+            if (StringUtils.isNotBlank(customLayoutFileRelativeBasePath)) {
+                productTitleFile = new File(getServletContext().getRealPath(customLayoutFileRelativeBasePath + "/product-title.jsp"));
+            }
+
+            if (productTitleFile != null && productTitleFile.exists()) {
+                productTitleFilePath = customLayoutFileRelativeBasePath + "/product-title.jsp";
+            } else {
+                productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
                 if (productTitleFile.exists()) {
+                    productTitleFilePath = "extensions/product-title.jsp";
+                } else {
+                    productTitleFilePath = "includes/product-title.jsp";
+                }
+            }
```

Comment:
WDYT :) ?

```suggestion
String productTitleFilePath = StringUtils.defaultIfBlank(
    customLayoutFileRelativeBasePath,
    "extensions/product-title.jsp"
);

if (!new File(getServletContext().getRealPath(productTitleFilePath)).exists()) {
    productTitleFilePath = "includes/product-title.jsp";
}
```

Reply:
I don't think this captures the entire logic that's needed here.
1. The file path that needs to be checked should be `customLayoutFileRelativeBasePath + "/product-title.jsp"` if `customLayoutFileRelativeBasePath` is not empty.
2. `customLayoutFileRelativeBasePath` being non-empty does not guarantee that `product-title.jsp` file is defined at that location, so we need to make sure to check the path `extensions/product-title.jsp` if it's not available at that location.

Reply:
Will use the following snippet derived from this code. Thanks for the suggestion.

```
String productTitleFilePath = "extensions/product-title.jsp";
if (StringUtils.isNotBlank(customLayoutFileRelativeBasePath)) {
    productTitleFilePath = customLayoutFileRelativeBasePath + "/product-title.jsp";
}
if (!new File(getServletContext().getRealPath(productTitleFilePath)).exists()) {
    productTitleFilePath = "includes/product-title.jsp";
}
```